src/details/ArborX_DetailsTreeNodeLabelling.hpp

```diff
@@ -22,64 +22,69 @@ namespace Experimental
 {
 
 template <class ExecutionSpace, class BVH, class LabelsIn, class LabelsOut>
-void initBVHLabels(ExecutionSpace const &exec_space, BVH const &bvh,
-                   LabelsIn const &in, LabelsOut &out)
+void initLabels(ExecutionSpace const &exec_space, BVH const &bvh,
+                LabelsIn const &in, LabelsOut &out)
 {
   auto const n = bvh.size();
 
+  ARBORX_ASSERT(n >= 2);
   ARBORX_ASSERT(in.size() == n);
   ARBORX_ASSERT(out.size() == 2 * n - 1);
 
   Kokkos::parallel_for(
-      "ArborX::Experimental::init_labels",
+      "ArborX::initialize_leaf_node_labels",
       Kokkos::RangePolicy<ExecutionSpace>(exec_space, n - 1, 2 * n - 1),
       KOKKOS_LAMBDA(int i) {
         out(i) = in(Details::HappyTreeFriends::getLeafPermutationIndex(bvh, i));
       });
 }
 
-template <class ExecutionSpace, class BVH, class BVHParents>
-void findBVHParents(ExecutionSpace const &exec_space, BVH const &bvh,
-                    BVHParents const &bvh_parents)
+template <class ExecutionSpace, class BVH, class Parents>
+void findParents(ExecutionSpace const &exec_space, BVH const &bvh,
+                 Parents const &parents)
 {
   auto const n = bvh.size();
 
-  ARBORX_ASSERT(bvh_parents.size() == 2 * n - 1);
+  ARBORX_ASSERT(n >= 2);
```

Comment:
Still think that `n = 1` is a valid case and we should just set the parents array to -1 in that case. In short, I think all the functions should work with any tree, including empty and single leaf.

Reply:
Need to think about it